src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +19,41 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't create phone: " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> criteriaQuery = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> phone = criteriaQuery.from(Phone.class);
+            Predicate predicate = criteriaBuilder.and();
+            for (Map.Entry<String, String[]> element : params.entrySet()) {
+                predicate = criteriaBuilder.and(predicate,
+                        phone.get(element.getKey()).in((Object[]) element.getValue()));
```

Comment:
```suggestion
                        phone.get(element.getKey()).in(element.getValue()));
```
can we go without this cast?